### ISSUE TITLE
Documentation: signatures with typehints.

### ISSUE THREAD

**Issue by pfebrer** [zerothi/sisl#638]
Function signatures with typehints in the documentation can become very hard to read:

![Screenshot from 2023-11-01 17-50-32](https://github.com/zerothi/sisl/assets/42074085/52374563-ac4c-4f0e-9f25-5628dfbc1cae)

I been playing with some CSS, and it is possible to have each parameter in a new line, just as you would do in the source code:

![Screenshot from 2023-11-01 18-11-13](https://github.com/zerothi/sisl/assets/42074085/e57ccccf-1044-448c-8bc3-4881e01b7257)

Do you think it would be better?

**Comment by pfebrer** [comment]
You can tune it further to make it even clearer by adding colors for example:

![Screenshot from 2023-11-01 18-20-48](https://github.com/zerothi/sisl/assets/42074085/dbdf7268-d6ac-46a5-838c-d75f6885c71e)

**Comment by pfebrer** [comment]
Or even with the weight of the font, here I avoid setting the typehint to bold:

![Screenshot from 2023-11-01 18-23-45](https://github.com/zerothi/sisl/assets/42074085/b346d5bd-b7e5-4cd2-96f7-31f901d627a0)

So, really you can do anything you come up with.

The way I did it applies the styling to methods as well, but I'm sure there's some way to avoid it:

![Screenshot from 2023-11-01 18-23-38](https://github.com/zerothi/sisl/assets/42074085/65bbc219-e0b0-4ac0-9648-49ef31f43839)

**Comment by pfebrer** [comment]
Perhaps we can do a poll on discord to see which layout users prefer :)

**Comment by zerothi** [comment]
I really like the grayed out typing construct.
I agree the typing is horrendous, I thought it would help clarify how the arguments are allowed, but it is just too much, see here: https://zerothi.github.io/sisl/api/generated/sisl.Geometry.html#sisl.Geometry.auc2sc
I really would like it to just point to `AtomsLike`.

**Comment by pfebrer** [comment]
> I really would like it to just point to AtomsLike.

See the trick here: https://github.com/pandas-dev/pandas/issues/33025#issuecomment-699636759

I think it would be very useful for `AtomsLike` and `AtomsArgument`.

> I really like the grayed out typing construct.

This one: ?

![image](https://github.com/zerothi/sisl/assets/42074085/195599e0-2ee7-4edc-899d-822a27a1fa1d)

**Comment by zerothi** [comment]
Great reference for hiding the information, lets do something like that. I'll play later.

As for the type hinting, I kind of prefer how it is shown in `Geometry`, no type-hints at the interface (for clarity), and an extensive type declaration in the `Parameters` section.
I am not fully sure why your `PdosPlot` shows differently from `Geometry`...

**Comment by pfebrer** [comment]
Don't you think the typehinted signature is clear enough if you give one line to each parameter? I like it because it gives you a quick idea of what the argument expects, as in the source code. 

I think this:

![Screenshot from 2023-11-01 18-23-45](https://github.com/zerothi/sisl/assets/42074085/e8f2c0f1-4742-4763-969a-8218fc8a4cea)

Is more helpful than this:

![Screenshot from 2023-11-02 09-58-41](https://github.com/zerothi/sisl/assets/42074085/b2227ee0-6151-4811-8d54-bf5aed1c61ec)

(I know the typing is missing in the argument description)

**Comment by zerothi** [comment]
The typing should be present in the argument description, I don't mind having them in the interface, IFF they are simple lines, the `Geometry` example is really bad since there the typing gets excessively long. If it hinders readability, I don't want it ;)
Users don't really care, and if it becomes difficult to see the order of arguments, it becomes hard.
So in that case the 1-line per argument is great!

**Comment by pfebrer** [comment]
Yeah I agree, I would like that the typehints are shown in the signature with 1 line per argument only if it is possible to use the trick to avoid expanding the complex typehints.

I think for users that are already familiar with type hints that might help, and also it shows that the package is type hinted, which is not clear if the typehint appears only in the description.
